Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -23,6 +25,7 @@ namespace MultiplayerSample
     AZ_CVAR(float, cl_WeaponsDrawDebugDurationSec, 10.0f, nullptr, AZ::ConsoleFunctorFlags::Null, "The number of seconds to display debug draw data");
     AZ_CVAR(float, sv_WeaponsImpulseScalar, 750.0f, nullptr, AZ::ConsoleFunctorFlags::Null, "A fudge factor for imparting impulses on rigid bodies due to weapon hits");
     AZ_CVAR(float, sv_WeaponsStartPositionClampRange, 1.f, nullptr, AZ::ConsoleFunctorFlags::Null, "A fudge factor between the where the client and server say a shot started");
+    AZ_CVAR(float, sv_WeaponsMaxRange, 25.f, nullptr, AZ::ConsoleFunctorFlags::Null, "Max range a weapon shot will travel during physics raycast.");
```

Comment:
This is actually max aim distance, and not max weapon distance?

It's been a long time, but the basic idea for third person weapon aiming is we do the raycast from the center of the screen through the target reticle and see what that hits, then have the character do some fudgery so that it aims at that exact same point in third person view, hence the todo. This will help compensate for error in the third person camera while aiming since we can't just perfectly aim down the gunsight like in first person.

The actual weapon travel distance is controlled by the weapon itself, so I might update this to sv_WeaponsMaxAimRange and possibly even make it a designer tunable archetype property on the weapons component, so they can tune better for short vs long-range weapons.

Reply:
If we're worried about first person mechanics at all, the third person aim algorithm described perfectly degrade to first person, as the raycast will be exactly down the gunsight anyways.

Reply:
What you're describing is what's happening. We raycast from the center of the screen to locate a target up to a max distance. Making it a component property is a good idea. Might as well have it be set per weapon.